Multisig/Data/Services/Safe Client Gateway Service/SCGModels.swift

```diff
@@ -0,0 +1,380 @@
+    let parameters: [SCGParameter]?
+}
+
+struct SCGParameter: Decodable {
+    let name: String
+    let type: String
+    let value: SCGParamValue
+}
+
+extension SCGParameter {
+    enum Key: String, CodingKey {
+        case name, type, value
+    }
+
+    init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: Key.self)
+        name = try container.decode(String.self, forKey: .name)
+        type = try container.decode(String.self, forKey: .type)
+        value = try container.decode(SCGParamValueWrapper.self, forKey: .value).value
+    }
+}
+
+protocol SCGParamValue {}
+
+extension String: SCGParamValue {}
+
+extension Array: SCGParamValue where Element == SCGParamValue {}
+
+enum SCGParamValueWrapper: Decodable {
+    case stringValue(String)
+    case arrayValue([SCGParamValueWrapper])
+
+    var value: SCGParamValue {
+        switch self {
+        case .stringValue(let value):
+            return value
+        case .arrayValue(let array):
+            return array.map { $0.value }
+        }
+    }
+
+    init(from decoder: Decoder) throws {
+        let container = try decoder.singleValueContainer()
+        if let string = (try? container.decode(String.self)) {
+            self = .stringValue(string)
+        } else {
+            self = try .arrayValue(container.decode([SCGParamValueWrapper].self))
+        }
+    }
+}
+
+// MARK: - Details
+
+struct SCGTransactionDetails: Decodable {
+    let executedAt: Date
+    let txStatus: SCGTransactionStatus
+    let txInfo: SCGTransactionInfo
+    let txData: SCGTransactionData
+    let detailedExecutionInfo: SCGDetailedExecutionInfo?
+    let txHash: DataString?
+}
+
+extension SCGTransactionDetails {
+    enum Key: String, CodingKey {
+        case executedAt, txStatus, txInfo, txData, detailedExecutionInfo, txHash
+    }
+
+    init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: Key.self)
+        executedAt = try container.decode(Date.self, forKey: .executedAt)
+        txStatus = try container.decode(SCGTransactionStatus.self, forKey: .txStatus)
+        txInfo = try container.decode(SCGTransactionInfoWrapper.self, forKey: .txInfo).value
+        txData = try container.decode(SCGTransactionData.self, forKey: .txData)
+        detailedExecutionInfo = try container.decodeIfPresent(SCGDetailedExecutionInfoWrapper.self, forKey: .detailedExecutionInfo)?.value
+        txHash = try container.decodeIfPresent(DataString.self, forKey: .txHash)
+    }
+}
+
+struct SCGTransactionData: Decodable {
+    let hexData: DataString?
+    let dataDecoded: SCGDataDecoded?
+    let to: AddressString
+    let value: UInt256String
+    let operation: SCGOperation
+}
+
+protocol SCGDetailedExecutionInfo: Decodable {}
+
+enum SCGDetailedExecutionInfoType: String, Decodable {
+    case MODULE
+    case MULTISIG
+}
+
+enum SCGDetailedExecutionInfoWrapper: Decodable {
+    case Module(SCGModuleExecutionDetails)
```

Comment:
Why upper case ?

Reply:
This is how it is in the safe client gateway, I went on with 1-to-1 case names wherever they are. Make it Swifty?